skin/light/light.css

```diff
@@ -241,6 +241,10 @@
   border-left: 4px solid #0996f8 !important;
 }
 
+.tabbrowser-tab[pinned] + .tabbrowser-tab:not([pinned]) {
+  border-top: 1px solid black !important;
+}
```

Comment:
During the drag, the border follows the tab being dragged. Once the drag is finished, the border goes to the newly first unpinned tab.